dask/dataframe/core.py

```diff
@@ -3693,7 +3702,10 @@ def _raise_if_object_series(x, funcname):
     if isinstance(x, Series) and hasattr(x, "dtype"):
         if x.dtype == object:
             raise ValueError("`%s` not supported with object series" % funcname)
-        elif pd.api.types.is_dtype_equal(x.dtype, "string"):
+        elif not PANDAS_GT_200 and pd.api.types.is_dtype_equal(x.dtype, "string"):
+            # Bug in pandas < 2.0 interprets decimal as string
```

Comment:
IIUC this is say "Use `pd.api.types.is_string_dtype` in pandas=2+, but `pd.api.types.is_dtype_equal(..., "string")` otherwise because there's a known bug". If that's the case, it'd be nice to have a little `is_string_dtype` utility in `dask.dataframe._comat` that handles that dispatching for us (based on what version of `pandas` is installed). 

Also, I don't mean this as a blocking comment on this PR. What you have here is totally fine. I could just see us reaching for `pd.api.types.is_string_dtype` in the future and using it w/o thinking about the bug.

Reply:
Created a utility function